From 24b6f917065daa83da2446638205b367fba5c335 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:12:21 -0400
Subject: [alsa] hda: Disable Front HP jack detection on Gigabyte Z87X-UD3H

Message-id: <140972823573-12-patchset-jkysela@redhat.com>
Patchwork-id: 90712
O-Subject: [RHEL 7.1 PATCH 011/442] ALSA: hda - Disable Front HP jack detection on Gigabyte Z87X-UD3H
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

This motherboard seems to have a flaky jack detection - when the
front HP is not present, the jack state quickly switches on and off.

This has been reported by three people in the bug, so I doubt it's
a user error this time.

BugLink: https://bugs.launchpad.net/bugs/1248116
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: b2c53e206967d01fd4fb6dd525f89ae738beb2e6)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 16c0caa..6a705a8 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -1810,6 +1810,7 @@ enum {
  ALC882_FIXUP_ACER_ASPIRE_7736,
  ALC882_FIXUP_ASUS_W90V,
  ALC889_FIXUP_CD,
+ ALC889_FIXUP_FRONT_HP_NO_PRESENCE,
  ALC889_FIXUP_VAIO_TT,
  ALC888_FIXUP_EEE1601,
  ALC882_FIXUP_EAPD,
@@ -2027,6 +2028,15 @@ static const struct hda_fixup alc882_fixups[] = {
    { }
   }
  },
+ [ALC889_FIXUP_FRONT_HP_NO_PRESENCE] = {
+  .type = HDA_FIXUP_PINS,
+  .v.pins = (const struct hda_pintbl[]) {
+   { 0x1b, 0x02214120 }, /* Front HP jack is flaky, disable jack detect */
+   { }
+  },
+  .chained = true,
+  .chain_id = ALC889_FIXUP_CD,
+ },
  [ALC889_FIXUP_VAIO_TT] = {
   .type = HDA_FIXUP_PINS,
   .v.pins = (const struct hda_pintbl[]) {
@@ -2258,7 +2268,7 @@ static const struct snd_pci_quirk alc882_fixup_tbl[] = {
  SND_PCI_QUIRK(0x1071, 0x8258, "Evesham Voyaeger", ALC882_FIXUP_EAPD),
  SND_PCI_QUIRK(0x1462, 0x7350, "MSI-7350", ALC889_FIXUP_CD),
  SND_PCI_QUIRK_VENDOR(0x1462, "MSI", ALC882_FIXUP_GPIO3),
- SND_PCI_QUIRK(0x1458, 0xa002, "Gigabyte EP45-DS3", ALC889_FIXUP_CD),
+ SND_PCI_QUIRK(0x1458, 0xa002, "Gigabyte EP45-DS3/Z87X-UD3H", ALC889_FIXUP_FRONT_HP_NO_PRESENCE),
  SND_PCI_QUIRK(0x147b, 0x107a, "Abit AW9D-MAX", ALC882_FIXUP_ABIT_AW9D_MAX),
  SND_PCI_QUIRK_VENDOR(0x1558, "Clevo laptop", ALC882_FIXUP_EAPD),
  SND_PCI_QUIRK(0x161f, 0x2054, "Medion laptop", ALC883_FIXUP_EAPD),
-- 
1.7.1